File: src/main/java/com/dochiri/hexagonal/application/user/dto/LoginUserResult.java
package com.dochiri.hexagonal.application.user.dto;

public record LoginUserResult(
        String id,
        String email,
        String role,
        String accessToken
) {
    public static LoginUserResult of(String id, String email, String role, String accessToken) {
        return new LoginUserResult(id, email, role, accessToken);
    }
}

File: src/main/kotlin/com/dochiri/hexagonal/application/user/dto/LoginUserResult.kt
package com.dochiri.hexagonal.application.user.dto

data class LoginUserResult(
    val id: String,
    val email: String,
    val role: String,
    val accessToken: String
) {
    companion object {
        fun of(id: String, email: String, role: String, accessToken: String): LoginUserResult =
            LoginUserResult(id, email, role, accessToken)
    }
}
